app/uk/gov/hmrc/customs/emailfrontend/connectors/Save4LaterConnector.scala

```diff
@@ -26,89 +26,103 @@ import uk.gov.hmrc.customs.emailfrontend.connectors.http.responses.{BadRequest,
 import uk.gov.hmrc.customs.emailfrontend.model.{EmailDetails, JourneyType, ReferrerName}
 import uk.gov.hmrc.http._
 import uk.gov.hmrc.http.HttpReads.Implicits._
+import uk.gov.hmrc.http.client.HttpClientV2
+
+import play.api.libs.ws.JsonBodyWritables.writeableOf_JsValue
 
 import scala.concurrent.{ExecutionContext, Future}
 
 @Singleton
-class Save4LaterConnector @Inject()(http: HttpClient,
+class Save4LaterConnector @Inject()(http: HttpClientV2,
                                     appConfig: AppConfig,
                                     audit: Auditable)(implicit ec: ExecutionContext) extends Logging {
 
   def getEmailDetails(id: String, key: String)
                      (implicit hc: HeaderCarrier, reads: Reads[EmailDetails]): Future[Option[EmailDetails]] = {
 
-    val url = s"${appConfig.save4LaterUrl}/$id/$key"
+    val getUrl = s"${appConfig.save4LaterUrl}/$id/$key"
```

Comment:
Better to rename it to endPointUrl or endPointUri or urlString as this name gives the feeling of method name.

Same for other methods of this class